### DISCUSSION TITLE
Speed as a % of filament max speed

### DISCUSSION THREAD

**Discussion by og54** [OrcaSlicer/OrcaSlicer#2310] [Ideas]
Hi!

I have a thought about having an option to make process print speeds as a percentage of the max speed setting in the filament settings.
This would make a user able to set up a print process where speeds would auto adjust for a given filament, and not just a max flow cap.

As I have no idea how to program something like this, I can just float the idea here and hope the right people sees it...

**Comment by houstnwehavuhoh** [comment]
YES! I would LOVE this. PrusaSlicer does this. You set your max volumetric flow, then max speed. From there, you can go into your layer speeds and basically say, "go as fast as my max allowed volumetric flow will allow, with respects to my max speed, HOWEVER, only go 75% of that speed for top infill layers" (or whatever you so choose, using top infill layers as an example).

I think this would be an awesome feature to implement. However, through the grape vine, there's word that in the next version of Orca, you won't even be able to to zero out your other layer speeds to allow speeds to be based off your max vol flow, you'll have to actually enter speeds. This is kind if moving the other direction IMO. Speculation though of course, we'll see

**Comment by AbbyCottontail** [comment]
Agreed, it would be most helpful, especially if one could set a max print speed on a per filament basis, then have it populate into the main settings based on the percentages. 

I envision the UI as allowing you to enter either  or mm/s, and if you've entered % it shows the mm/s beside the entry.

**Comment by gshpychka** [comment]
This might be confusing for the user since the max speed depends on the max flow rate, line width, and layer height. So e.g. with variable layer height the speeds will vary per layer.
